lua/nvim-treesitter/refactor/highlight_definitions.lua

```diff
@@ -0,0 +1,75 @@
+-- This module highlights reference usages and the corresponding
+-- definition on cursor hold.
+
+local parsers = require'nvim-treesitter.parsers'
+local ts_utils = require'nvim-treesitter.ts_utils'
+local locals = require'nvim-treesitter.locals'
+local api = vim.api
+local cmd = api.nvim_command
+
+local M = {}
+
+local usage_namespace = api.nvim_create_namespace('nvim-treesitter-usages')
+
+function M.highlight_usages(bufnr)
+  M.clear_usage_highlights(bufnr)
+
+  local node_at_point = ts_utils.get_node_at_cursor()
+
+  if not node_at_point then return end
+
+  local def_node, scope = ts_utils.find_definition(node_at_point, bufnr)
```

Comment:
Maybe we could only look for a definition when `node_at_point` is a `@reference`. This would prevent looking for definitions when there is not reference like kw-params in python. Just an idea, since the current solution is very light weight and fast.

![image](https://user-images.githubusercontent.com/7189118/85931347-12660c80-b8c4-11ea-8e9b-4b424ccda350.png)

Reply:
done

Reply:
I had the same problem with Typescript types.